server/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -97,6 +101,15 @@ public PluginInfo(final StreamInput in) throws IOException {
         this.name = in.readString();
         this.description = in.readString();
         this.version = in.readString();
+        if (in.getVersion().onOrAfter(Version.V_6_3_0)) {
+            elasticsearchVersion = Version.readVersion(in);
+            javaVersion = in.readString();
+        } else {
+            // the plugin must have the version of whichever node we are talking to, since this is enforced on startup
+            elasticsearchVersion = in.getVersion();
+            // this might not be true, but it is not important, we just need something here for bwc that is a valid java version string
+            javaVersion = System.getProperty("java.version");
```

Comment:
Or maybe hardcode this to what a JVM running our minimum supported version would report ("1.8")?